Add title component

hirundinidae opened the pull request:
## [EASY] Title UAS
Adds title component with title, description, image, background color, text alignment, back arrow as props. 

resources.jsx (/resources) can be used to test different props, will be changed to the actual "Resources" title in next PR 

![screenshot_20210801_205242](https://user-images.githubusercontent.com/65516245/127791256-e5d81d1a-450b-4981-98d5-1065a5bc5a19.png)
![screenshot_20210801_205258](https://user-images.githubusercontent.com/65516245/127791254-8433ba8b-5dd2-4640-a09a-a5eff0f125c9.png)
![screenshot_20210801_205341](https://user-images.githubusercontent.com/65516245/127791275-cf8952a5-3f67-4fbd-a2df-a5fedc29461c.png)
![screenshot_20210801_205327](https://user-images.githubusercontent.com/65516245/127791276-cebe1a7e-80c1-4073-a384-46ad0c7daf57.png)

hirundinidae committed:
Add title section components

hirundinidae committed:
Add props

hirundinidae committed:
Link button and add description

hirundinidae committed:
Add left alignment prop and image

hirundinidae committed:
Remove unneeded styling

hirundinidae committed:
Add styles for mobile

hirundinidae committed:
Add props comment

hirundinidae committed:
Fix position of back arrow

hirundinidae committed:
Clean up

hirundinidae committed:
Add image height and width props

xsharonhe commented on components/Title.jsx:
```diff
@@ -0,0 +1,127 @@
+import React from "react";
```
This line isn't needed in Next because every page automatically uploads React

hirundinidae replied:
Yep, mb

xsharonhe commented on components/Title.jsx:
```diff
@@ -0,0 +1,127 @@
+import { Text } from "../components";
+import { media } from "../utils";
+
+/**
```
Much better documentation I guess I will start converting mine to this from now on. :)

hirundinidae replied:
Haha, added it because there's a lot of props

xsharonhe commented on pages/resources.jsx:
```diff
@@ -9,6 +11,12 @@ export default function About({ ...props }) {
         <title>PuMP</title>
       </Head>
       <PageLayout>
+        <Title
```
Example is on the wrongg page haha can we move this to the uas.jsx file under pages/resources folder. Also can you help me rename the function to Resources instead of About

hirundinidae replied:
Yep, just tossed it here to test. Will add the titles to their pages according to available designs.

hirundinidae replied:
Removed in next commit, I'll make another PR putting all the titles on pages

xsharonhe commented on pages/resources.jsx:
```diff
@@ -1,6 +1,8 @@
+import { Title } from "../components";
+import SampleGraphic from "../public/home/impact-section-graphic-3.svg";
 
 export default function About({ ...props }) {
```
This is my mistake. Could you help me change this to resources?

xsharonhe commented on components/Title.jsx:
```diff
@@ -0,0 +1,127 @@
+const Wrapper = styled.div`
+  display: flex;
+  ${({ backgroundColor }) => `
+    background-color: ${backgroundColor};
```
Could you add a TODO here because we'll probably have this replaced by one of the blobs from design

xsharonhe reviewed (dismissed):
Nice overall! And for taking multiple use cases for reusability.

hirundinidae committed:
Make changes outlined in review

- remove unneeded lines
- clean up testing on resources.jsx
- add TODO reminder for background

xsharonhe reviewed (approved):
LGTM
